src/crypto/shelley/ShelleyWallet.js

```diff
@@ -153,32 +143,26 @@ export default class ShelleyWallet extends Wallet implements WalletInterface {
     Logger.info(`create wallet (implementationId=${String(implementationId)})`)
     this.id = uuid.v4()
     assert.assert(!this.isInitialized, 'createWallet: !isInitialized')
+    assert.assert(
+      isByron(implementationId) || isHaskellShelley(implementationId),
+      'ShelleyWallet::create: invalid walletImplementationId',
+    )
     const masterKey = await byronUtil.getMasterKeyFromMnemonic(mnemonic)
     await this.encryptAndSaveMasterKey(
       'MASTER_PASSWORD',
       masterKey,
       newPassword,
     )
-    let _purpose
-    switch (implementationId) {
-      case WALLET_IMPLEMENTATION_REGISTRY.HASKELL_SHELLEY:
-        _purpose = CONFIG.NUMBERS.WALLET_TYPE_PURPOSE.CIP1852
-        break
-      case WALLET_IMPLEMENTATION_REGISTRY.HASKELL_BYRON:
-        _purpose = CONFIG.NUMBERS.WALLET_TYPE_PURPOSE.BIP44
-        break
-      default:
-        throw new Error('wallet implementation id not valid')
-    }
+    const purpose = isByron(implementationId)
+      ? CONFIG.NUMBERS.WALLET_TYPE_PURPOSE.BIP44
+      : CONFIG.NUMBERS.WALLET_TYPE_PURPOSE.CIP1852
 
     const masterKeyPtr = await Bip32PrivateKey.from_bytes(
       Buffer.from(masterKey, 'hex'),
     )
-    const accountKey = await (await (await masterKeyPtr.derive(
-      _purpose,
-    )).derive(CONFIG.NUMBERS.COIN_TYPES.CARDANO)).derive(
-      0 + CONFIG.NUMBERS.HARD_DERIVATION_START,
-    )
+    const accountKey = await (await (await masterKeyPtr.derive(purpose)).derive(
+      CONFIG.NUMBERS.COIN_TYPES.CARDANO,
+    )).derive(0 + CONFIG.NUMBERS.HARD_DERIVATION_START)
```

Comment:
be careful about hard-coding the 0th account, since we'll have to support multiple accounts per wallet some time this year.